zipline/gens/brokers/ib_broker.py

```diff
@@ -500,12 +510,21 @@ def subscribed_assets(self):
 
     def subscribe_to_market_data(self, asset):
         if asset not in self.subscribed_assets:
+            log.debug("Subscribing to market data for {}".format(
+                asset))
+
             # remove str() cast to have a fun debugging journey
             self._tws.subscribe_to_market_data(str(asset.symbol))
             self._subscribed_assets.append(asset)
-
-            while asset.symbol not in self._tws.bars:
+	    counter = 0
+            while asset.symbol not in self._tws.bars and counter < _max_wait_subscribe: # waiting X secs
```

Comment:
Please use the 'polling' library for this purpose. It would make the clean much cleaner:
https://pypi.python.org/pypi/polling/0.3.0

Reply:
I'll look at it. could be cleaner indeed